### PR TITLE
Update deprecation comments in setup.py

### LINKED ISSUE #12744: Difference of extras Airflow 2.0 vs. Airflow 1.10
**Description**

When airflow 2.0 is installed from  PyPI, providers are not installed by default. In order to install them, you should add an appropriate extra. While this behavior is identical in Airflow 1.10 for those "providers" that required additional packages, there were a few "providers" that did not require any extras to function (example http, ftp) - we have "http", "ftp" extras for them now, but maybe some of those are popular enough to be included by default?.

We have to make a decision now:

- [x] should all of them (or some of them) be included by default when you install Airflow?
- [x] if we decide to exclude only some (or none), we should add them in UPGRADING_to_2_0 and in UPDATING documentation.

**Use case / motivation**

We want people to get a familiar experience when installing airflow. Why we provide familiar mechanism (with extras) and people will expect a slightly different configurations, installation and we can describe the differences, maybe some of those providers are so popular that we should include them by default? 

**Related Issues**

#12685 - where we discuss which of the extras should be included in the Production Image of 2.0.

**Additional info**

Here is the list of all "providers" that were present in 1.10 and had no additional dependencies - so basically they woudl work out-fhe-box in 1.10, but they need appropriate "extra" in 2.0.

*  "apache.pig": [],
*  "apache.sqoop": [],
*  "dingding": [],
*  "discord": [],
*  "ftp": [],
*  "http": [],
*  "imap": [],
*  "openfaas": [],
*  "opsgenie": [],
*  "sqlite": [],

Also here I appeal to the wisdom of crowd: @ashb, @dimberman @kaxil, @turbaszek, @mik-laj. @XD-DENG, @feluelle, @eladkal, @ryw, @vikramkoka, @KevinYang21  - let me know WDYT before I bring it to devlist?

### PR DESCRIPTION
**Pull request by potiuk**
Updates comments in setup.py according to the latest 2.0 agreements
about semver and clarifies use for arrays defined in setup.py.
    
We will refactor those after 2.0 further, but for now we keep
the current structure.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Added the 'pre-installed providers' we agreed to . Unfortunately @ashb it looks like we have to go back with install_requires to setup.py because depending on dev/prod environment (INSTALL_PROVIDERS_FROM_SOURCES) we have to dynmically add them to install_requires (unless of course we figure our something else). Seems like install_requires in setup.py overrides the one that is in setup.cfg unfortunately.

**Comment by ashb** [comment]
> Added the 'pre-installed providers' we agreed to . Unfortunately @ashb it looks like we have to go back with install_requires to setup.py because depending on dev/prod environment (INSTALL_PROVIDERS_FROM_SOURCES) we have to dynmically add them to install_requires (unless of course we figure our something else). Seems like install_requires in setup.py overrides the one that is in setup.cfg unfortunately.

Let me have a look - see if there's anything we can do

**Comment by ashb** [comment]
Is the only case we have to deal with when you do `pip install -e .` right?

**Comment by ashb** [comment]
See https://github.com/apache/airflow/pull/12916 for an approach that things in setup.cfg

**Comment by potiuk** [comment]
Yep, #12916 does it much better ! I turned this one into comment-only change @ashb

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -241,7 +241,7 @@ repos:
       - id: setup-order
         name: Check order of dependencies in setup.py
         language: python
-        files: ^setup.py$
+        files: ^setup.py$|^setup.cfg$
```
Comment (ashb) [inline .pre-commit-config.yaml:244]: Don't need this anymore do we?

**Review by ashb — APPROVED** [review]
Other than the pre-commit this LGTM.

Pre-emptively approving.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (potiuk) [reply .pre-commit-config.yaml:244]: We do. There is a check that checks sequence in setup.cfg as well :)

Comment (potiuk) [reply .pre-commit-config.yaml:244]: You've updated it when you moved the "install_requires".

Comment (potiuk) [reply .pre-commit-config.yaml:244]: But the comment + description of this commit + the files spec were not updated then. I modified the description now to be clear that 'setup-order' concerns both .py and .cfg

**Pull request merged**